checks/raw/pinned_dependencies.go

```diff
@@ -767,32 +779,36 @@ var validateGitHubActionWorkflow fileparser.DoWhileTrueOnFileContent = func(
 			if strings.HasPrefix(execAction.Uses.Value, "./") {
 				continue
 			}
-
-			dep := checker.Dependency{
-				Location: &checker.File{
-					Path:      pathfn,
-					Type:      finding.FileTypeSource,
-					Offset:    uint(execAction.Uses.Pos.Line),
-					EndOffset: uint(execAction.Uses.Pos.Line), // `Uses` always span a single line.
-					Snippet:   execAction.Uses.Value,
-				},
-				Pinned: asBoolPointer(isActionDependencyPinned(execAction.Uses.Value)),
-				Type:   checker.DependencyUseTypeGHAction,
-			}
-			parts := strings.SplitN(execAction.Uses.Value, "@", 2)
-			if len(parts) > 0 {
-				dep.Name = asPointer(parts[0])
-				if len(parts) > 1 {
-					dep.PinnedAt = asPointer(parts[1])
-				}
-			}
+			dep := createUnpinnedDep(execAction.Uses.Value, pathfn, execAction.Uses.Pos.Line)
 			pdata.Dependencies = append(pdata.Dependencies, dep)
 		}
 	}
 
 	return true, nil
 }
 
+func createUnpinnedDep(uses, pathfn string, line int) checker.Dependency {
```

Comment:
i have some confusion around the name.
1. This is specifically for actions, since it calls `isActionDependencyPinned`
2. We don't know ahead of time whether it's pinned or unpinned

Perhaps something like `newGHActionDependency`? or similar?